src/observable.js

```diff
@@ -96,6 +96,13 @@ function observable(value) {
       return nextValue;
     }
 
+    // do not set the same value again
+    if (typeof comparator === 'function') {
+      if (comparator(nextValue,value) == true) { return value }
+    } else {
+      if (nextValue === value) { return value }
+    }
```

Comment:
this can be done outside the library code, I don't to plan to add this inside.

Reply:
well, in that case we should close this pull request as I need this behaviour myself - which means that I will have to maintain a separate branch with this functionality